langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithJsonSchemaIT.java

```diff
@@ -90,6 +93,245 @@ void should_extract_pojo_with_primitives() {
+
+    @Test
+    void should_extract_set_of_pojo() {
+
+        for (ChatLanguageModel model : models()) {
+
+            // given
+            model = spy(model);
+
+            PersonSetExtractor personExtractor = AiServices.create(PersonSetExtractor.class, model);
+
+            String text = "Klaus is 37 years old, 1.78m height and single. " +
+                    "Franny is 35 years old, 1.65m height and single.";
+
+            // when
+            Set<PersonSetExtractor.Person> people = personExtractor.extractPersonFrom(text);
+
+            // then
+            assertThat(people).hasSize(2);
+
+            // Use assertThat with filters to ensure each person is in the set
+            assertThat(people).anyMatch(person ->
+                    person.name.equals("Klaus") &&
+                            person.age == 37 &&
+                            person.height.equals(1.78) &&
+                            !person.married
+            );
+
+            assertThat(people).anyMatch(person ->
+                    person.name.equals("Franny") &&
+                            person.age == 35 &&
+                            person.height.equals(1.65) &&
+                            !person.married
+            );
+
+            verify(model).chat(ChatRequest.builder()
+                    .messages(singletonList(userMessage(text)))
+                    .responseFormat(ResponseFormat.builder()
+                            .type(JSON)
+                            .jsonSchema(JsonSchema.builder()
+                                    .name("Set_of_Person")
+                                    .rootElement(JsonObjectSchema.builder()
+                                            .addProperty("array", JsonArraySchema.builder()
+                                                    .items(JsonObjectSchema.builder()
+                                                            .addStringProperty("name")
+                                                            .addIntegerProperty("age")
+                                                            .addNumberProperty("height")
+                                                            .addBooleanProperty("married")
+                                                            .required("name", "age", "height", "married")
+                                                            .build())
+                                                    .build())
+                                            .required("array")
+                                            .build())
+                                    .build())
+                            .build())
+                    .build());
+            verify(model).supportedCapabilities();
+        }
+    }
+
+    interface MaritalStatusExtractor {
+
+        enum MaritalStatus {
+
+            SINGLE, MARRIED
+        }
+
+        List<MaritalStatus> extractMaritalStatusFrom(String text);
+    }
+
+    @Test
+    void should_extract_list_of_enums() {
```

Comment:
please add one more test for set of enums

Reply:
implementation + test = done :)